clients/src/main/java/org/apache/kafka/clients/consumer/internals/DefaultBackgroundThread.java

```diff
@@ -109,34 +129,31 @@ public void run() {
                 try {
                     runOnce();
                 } catch (final WakeupException e) {
-                    log.debug(
-                        "Exception thrown, background thread won't terminate",
-                        e
-                    );
-                    // swallow the wakeup exception to prevent killing the
-                    // background thread.
+                    log.debug("WakeupException caught, background thread won't be interrupted");
+                    // swallow the wakeup exception to prevent killing the background thread.
                 }
             }
         } catch (final Throwable t) {
-            log.error(
-                "The background thread failed due to unexpected error",
-                t
-            );
-            if (t instanceof RuntimeException)
-                this.exception.set(Optional.of((RuntimeException) t));
-            else
-                this.exception.set(Optional.of(new RuntimeException(t)));
+            log.error("The background thread failed due to unexpected error", t);
+            throw new RuntimeException(t);
         } finally {
             close();
             log.debug("{} closed", getClass());
         }
     }
 
     /**
-     * Process event from a single poll
+     * Process event from a single poll. It performs the following tasks sequentially:
+     *  1. Try to poll and event from the queue, and try to process it.
+     *  2. Try to find Coordinator if needed
+     *  3. Try to send fetches.
+     *  4. Poll the networkClient for outstanding requests. If non: poll until next
+     *  iteration.
      */
     void runOnce() {
+        // TODO: we might not need the inflightEvent here
```

Comment:
Can we just get rid of it if we're not using it yet?

Reply:
Can we leave it to the next "minor" PR for the clean up? (per previous discussion)

Reply:
I'd be inclined to just drop it here if we are already adding a TODO.